blackjax/sgmcmc/diffusion.py

```diff
@@ -35,6 +35,102 @@ def one_step(
 
     return one_step
 
+""" 
+    We are interested in the simulations of exp(-U(x) / T), 
+    where U is energy function and T is the temperature.
+
+    logprob normally means U(x)/ T instead of U(x).
+"""
+
+
+class ContourState(NamedTuple):
+    position: PyTree
+    logprob_grad: PyTree
+    energy_pdf: PyTree
+    energy_idx: int
+
+
+def logposterior_estimator_fn(position: PyTree, data_batch: PyTree) -> float:
```

Comment:
I will refactor this out of `sgmcmc.gradients` in #299. Can you copy it there (even if that means it's implemented twice in that file for the time being?

Reply:
Gotcha will do!

Reply:
> I will refactor this out of `sgmcmc.gradients` in #299. Can you copy it there (even if that means it's implemented twice in that file) for the time being?

Could you elaborate on this part? I don't know what to do next.